p2p/client.go

```diff
@@ -16,36 +19,63 @@ import (
 	"github.com/libp2p/go-libp2p-peer"
 	pstore "github.com/libp2p/go-libp2p-peerstore"
 	ma "github.com/multiformats/go-multiaddr"
-	"google.golang.org/grpc"
 )
 
 // Client defines methods that can be implemented by any type handling p2p communications.
 type Client interface {
-	OpenClient(id identity.Identity) (p2ppb.P2PServiceClient, error)
+
+	// GetSignaturesForDocument gets the signatures for document
 	GetSignaturesForDocument(ctx *header.ContextHeader, identityService identity.Service, doc *coredocumentpb.CoreDocument) error
+
+	// after all signatures are collected the sender sends the document including the signatures
+	SendAnchoredDocument(ctx context.Context, id identity.Identity, in *p2ppb.AnchorDocumentRequest) (*p2ppb.AnchorDocumentResponse, error)
+}
+
+func (s *p2pServer) SendAnchoredDocument(ctx context.Context, id identity.Identity, in *p2ppb.AnchorDocumentRequest) (*p2ppb.AnchorDocumentResponse, error) {
+	pid, err := s.getPeerID(id)
+	if err != nil {
+		return nil, err
+	}
+	adr, err := proto.Marshal(in)
+	if err != nil {
+		return nil, err
+	}
+	recv, err := s.mes.sendRequest(ctx, pid, &protocolpb.P2PEnvelope{Type: protocolpb.MessageType_MESSAGE_TYPE_SEND_ANCHORED_DOC, Body: adr}, CentrifugeProtocol)
```

Comment:
About RPC:
yes it looks like an RPC system for ease of use(You could also call it distributed event handling :D ), but we have the freedom to do the wire protocol the way we want instead of dealing with http2 framing and other stuff that GRPC does.
I will fix the naming its confusing here, adr is not an address.